File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/adapter/api/client/dtos/errors/ProblemDetailResponse.java
package com.mycompany.myshop.testkit.driver.adapter.api.client.dtos.errors;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProblemDetailResponse {
    private String type;
    private String title;
    private Integer status;
    private String detail;
    private String instance;
    private String timestamp;
    private List<FieldError> errors;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class FieldError {
        private String field;
        private String message;
        private String code;
        private Object rejectedValue;
    }
}

File: system-test/dotnet/Driver.Adapter/Api/Client/Dtos/Errors/ProblemDetailResponse.cs
namespace Driver.Adapter.Api.Client.Dtos.Errors;

public class ProblemDetailResponse
{
    public string? Type { get; set; }
    public string? Title { get; set; }
    public int? Status { get; set; }
    public string? Detail { get; set; }
    public string? Instance { get; set; }
    public string? Timestamp { get; set; }
    public List<FieldError>? Errors { get; set; }

    public class FieldError
    {
        public string? Field { get; set; }
        public string? Message { get; set; }
        public string? Code { get; set; }
        public object? RejectedValue { get; set; }
    }
}

